src/Image/Point.php

```diff
@@ -38,6 +38,12 @@ final class Point implements PointInterface
      */
     public function __construct($x, $y)
     {
+        if (!\is_int($x)) {
```

Comment:
If we pass wrong arguments, they will be [silently converted to 0](https://3v4l.org/FKFSe).

I'd instead do something like

```php
        if (!is_numeric($x)) {
            throw new InvalidArgumentException('x must be numeric');
        }
        $x = (int) $x;
        if (!is_numeric($y)) {
            throw new InvalidArgumentException('y must be numeric');
        }
        $y = (int) $y;
```

Reply:
I think for better consistency, this should be handled the same way as `Box` class does: https://github.com/php-imagine/Imagine/blob/c10ea7f80e5b93473e2390515adc3057b85e26a2/src/Image/Box.php#L41-L46

Reply:
Ah, I didn’t think of the `0` case. 
So +1 for you latest code suggestion 👍